Sources/NIOCore/Utilities.swift

```diff
@@ -257,4 +257,14 @@ extension System {
     /// The option can be enabled by setting the ``ChannelOptions/Types/DatagramReceiveOffload`` channel option.
     public static let supportsUDPReceiveOffload: Bool = false
     #endif
+
+    /// Returns the UDP maximum segment count if the platform supports and defines it.
+    public static var udpMaxSegments: Int? {
+        #if os(Linux)
+        if CNIOLinux_UDP_MAX_SEGMENTS != -1 {
```

Comment:
Mild nit: I suspect the optimizer will end up calling this function twice, as it doesn't know what it does. I recommend that you save the value to a temporary and branch from there.

Reply:
I _think_ I've addressed this but your comment isn't 100% clear to me.